src/components/Filter/index.js

```diff
@@ -0,0 +1,100 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import { renderElement } from '../../util';
+import {
+  filtersStyles,
+  filterDropdownStyles,
+  filterIconStyles
+} from '../../styles/filter.styles';
+import { RendererType } from '../../constants';
+
+export default class Filter extends Component {
+  state = {
+    visible: false
+  };
+
+  componentDidMount() {
+    document.addEventListener('click', this.handleClose, true);
+    document.addEventListener('keyup', this.handleClose, true);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.handleClose, true);
+    document.removeEventListener('keyup', this.handleClose, true);
+  }
+
+  toggleDropdownVisibility = visible => {
+    this.setState(() => ({
+      visible
+    }));
+  };
+
+  handleMenuRef = ref => {
+    this.menu = ref;
+  };
+
+  handleIconRef = ref => {
+    this.icon = ref;
+  };
+
+  handleIconClick = () => {
+    if (this.state.visible) {
+      this.toggleDropdownVisibility(false);
+    } else {
+      this.toggleDropdownVisibility(true);
+    }
+  };
+
+  handleClose = e => {
+    if (this.state.visible) {
+      if (
+        e.key === 'Escape' ||
+        (this.icon &&
+          this.menu &&
+          e.target !== this.icon &&
+          !this.menu.contains(e.target))
+      ) {
+        this.toggleDropdownVisibility(false);
+      }
+    }
+  };
+
+  render() {
+    const { filterRenderer, data, dataKey } = this.props;
+    return (
+      <span>
```

Comment:
Rendering an empty `span` in case of filter renderer not being present is redundant.